packages/hasher/src/resolveInternalDependencies.ts

```diff
@@ -0,0 +1,59 @@
+import { PackageHashInfo } from "./hashOfPackage";
+import {
+  WorkspaceInfo,
+  listOfWorkspacePackageNames,
+  findWorkspacePath
+} from "./yarnWorkspaces";
+
+export type Dependencies = { [key in string]: string };
+
+export function filterInternalDependencies(
+  dependencies: Dependencies,
+  workspaces: WorkspaceInfo
+): string[] {
+  const workspacePackageNames = listOfWorkspacePackageNames(workspaces);
+  return Object.keys(dependencies).filter(
+    dependency => workspacePackageNames.indexOf(dependency) >= 0
+  );
+}
+
+function isDone(done: PackageHashInfo[], packageName: string): boolean {
+  return Boolean(done.find(({ name }) => name === packageName));
+}
+
+function isInQueue(queue: string[], packagePath: string): boolean {
+  return queue.indexOf(packagePath) >= 0;
+}
+
+export function addToQueue(
+  name: string,
```

Comment:
Can this queue abstraction be isolated, it seems to be used in multiple places.
Isolating it would also reduce couple; it seems that in multiple places we manually manage a queue by passing variables around.

Reply:
Thanks for pushing for this ❤️ I've moved the queue management inside `resolveInternalDependencies` out and into the module that deals with the queue.